---
title: Figure Captions
author: Joshua J. Hamilton
date: September 16, 2016
---

&nbsp;

## Figure 1 (figures/fig1-tree/tree-abbrev.pdf)

&nbsp;

Phylogenetic placement of the SAGs and MAGs within the acI lineage, relative to other sequenced actinobacterial genomes in the class Actinobacteria [@Gao2012]. The tree was built using RAxML [@Stamatakis2014] from a concatenated alignment of single-copy marker genes [@Darling2014]. The class Acidimicrobiia forms the outgroup.

&nbsp;

## Figure 2a(b) (figures/fig2-sampling/completeness-2.svg and figures/fig2-sampling/completeness-1.svg)

&nbsp;

Mean estimated completeness of tribe(clade)-level population genomes. For each tribe(clade), genomes were randomly sampled (with replacement) from the set of all genomes belonging to that tribe(clade). Completeness was estimated using single-copy marker genes [@Parks2015]. Error bars represent the 95% confidence estimated from 1000 iterations.

&nbsp;

## Supplementary Figure 1 (figures/fig1-tree/tree-full.pdf)

&nbsp;

Phylogenetic placement of the SAGs and MAGs within the acI lineage, relative to other sequenced actinobacterial genomes in the class Actinobacteria [@Gao2012]. The tree was built using RAxML [@Stamatakis2014] from a concatenated alignment of single-copy marker genes [@Darling2014]. The class Acidimicrobiia forms the outgroup.

&nbsp;

## Supplementary Figure 2 (figures/fig3-workflow/FigureS2.pdf)

&nbsp;

Pruning a metabolic network graph, using glycolysis as an example. (A) Metabolic reconstruction for glycolysis. This panel also shows the KBase reaction IDs corresponding to each step of glycolysis, as well as the gene(s) encoding the enzyme for that step. Genes are shown only for the TE02754 genome. (B) Glycolysis represented as a metabolic network graph. Currency metabolites are circumscribed by a dotted rectangle. For clarity, protons are not shown. (C) Representation of glycolysis after pruning. The images in (A) and (B) are modified from [@Ma2003].

&nbsp;

## Supplementary Figure 3 (figures/fig3-workflow/FigureS3.pdf)

&nbsp;

Identifying seed compounds in metabolic networks. (A) Metabolic network graph for an imaginary network. Grey circles represent metabolites and arcs represent reactions or parts of reactions. (B) To identify seed compounds, the metabolic network graph is first decomposed into its strongly connected components (SCCs), sets of nodes such that each node in the set is reachable from every other node. Here, each colored set of nodes corresponds to a unique SCC. (C) SCC decomposition enables seed sets to be identified from source components (components with no incoming edges) on the condensation of the original graph. In the condensation of the original graph shown here, each colored node corresponds to a unique SCC (and a potential seed set). Seed sets are enclosed in dotted circles. The pink seed set contains three seed compounds, the orange contains two, and the blue and yellow each contain one seed compound. The images in (A) and (B) are modified from [@Borenstein2008].

&nbsp;

## References

&nbsp;
