---
title: Figure Captions
author: Joshua J. Hamilton
date: September 16, 2016
---

&nbsp;

## Figure 1 (figures/fig1-tree/tree-abbrev.pdf)

&nbsp;

Phylogenetic placement of the SAGs and MAGs within the acI lineage, relative to other sequenced actinobacterial genomes in the class Actinobacteria [@Gao2012]. The tree was built using RAxML [@Stamatakis2014] from a concatenated alignment of single-copy marker genes [@Darling2014]. The class Acidimicrobiia forms the outgroup.

&nbsp;

## Figure 2a(b) (figures/fig2-sampling/completeness-2.svg and figures/fig2-sampling/completeness-1.svg)

&nbsp;

Mean estimated completeness of tribe(clade)-level population genomes. For each tribe(clade), genomes were randomly sampled (with replacement) from the set of all genomes belonging to that tribe(clade). Completeness was estimated using single-copy marker genes [@Parks2015]. Error bars represent the 95% confidence estimated from 1000 iterations.

&nbsp;

## Supplementary Figure 1 (figures/fig1-tree/tree-full.pdf)

&nbsp;

Phylogenetic placement of the SAGs and MAGs within the acI lineage, relative to other sequenced actinobacterial genomes in the class Actinobacteria [@Gao2012]. The tree was built using RAxML [@Stamatakis2014] from a concatenated alignment of single-copy marker genes [@Darling2014]. The class Acidimicrobiia forms the outgroup.

&nbsp;

## Supplementary Figure 2 (figures/fig3-workflow/FigureS2.pdf)

&nbsp;

Workflow for pruning a metabolic network reconstruction, using glycolysis as an example. (A) Canonical representation of glycolysis. This panel also shows the KBase reaction IDs corresponding to each step, as well as the gene(s) encoding the enzyme for that step. Genes are shown only for the TE02754 genome. (B) Glycolysis represented as a metabolic network graph. Currency metabolites are circumscribed by a dotted rectangle. For clarity, protons are not shown. (C) Representation of glycolysis after pruning. The images in (A) and (B) are modified from [@Ma2003].


&nbsp;

## References

&nbsp;
