---
title: Metabolic Network Analysis and Metatranscriptomics Reveals Auxotrophies and Nutrient Sources of the Cosmopolitan Freshwater Microbial Lineage acI
author: Joshua J. Hamilton^1\*^, Sarahi L. Garcia^2, Brittany S. Brown^1, Ben O. Oyserman^3, Francisco Moya-Flores^3, Stefan Bertilsson^2,4, Rex R. Malmstrom^5, Katrina T. Forest^1, Katherine D. McMahon^1,3
abstract: \* Correspondence&#58; Joshua J. Hamilton, jjhamilton2@wisc.edu
date: ^1^ Department of Bacteriology, University of Wisconsin-Madison, Madison, WI, USA; ^2^ Department of Ecology and Genetics, Uppsala University, Uppsala, Sweden; ^3^ Department of Civil and Environmental Engineering, University of Wisconsin-Madison, Madison, WI, USA; ^4^ Science for Life Laboratory, Uppsala University, Uppsala, Sweden; ^5^ United States Department of Energy Joint Genome Institute, Walnut Creek, CA, USA
---

# Abstract

An explosion in the number of available genome sequences obtained through metagenomics and single-cell genomics has enabled a new view of the diversity of microbial life, yet we know surprisingly little about how microbes interact with each other or their environment. In fact, the majority of microbial species remain uncultivated, while our perception of their ecological niches is based on reconstruction of their metabolic potential. In this work, we demonstrate how the "seed set framework", which computes the set of compounds that an organism must acquire from its environment [@Borenstein2008], enables computational analysis of metabolic reconstructions, while providing new insights into a microbe's metabolic capabilities, such as nutrient use and auxotrophies. We apply this framework to members of the ubiquitous freshwater Actinobacterial lineage acI, confirming and extending previous experimental and genomic observations implying that acI bacteria are heterotrophs reliant on peptides and saccharides. We also present the first metatranscriptomic study of the acI lineage, revealing high expression of transport proteins and the light-harvesting protein actinorhodopsin. Putative transport proteins complement predictions of nutrients and essential metabolites while providing additional support to the hypothesis that members of the acI are photoheterotrophs.

# Importance

The metabolic activity of uncultivated microorganisms contributes to numerous ecosystem processes, ranging from nutrient cycling in the environment to influencing human health and disease. Advances in sequencing technology have enabled the assembly of genomes for these microorganisms, but our ability to generate reference genomes far outstrips our ability to analyze them. Common approaches to analyzing microbial metabolism require reconstructing the entirety of an organism's metabolic pathways, or performing targeted searches for genes involved in a specific process. This paper presents a third approach, in which draft metabolic reconstructions are used to identify compounds through which an organism may interact with its environment. These compounds can then guide more intensive metabolic reconstruction efforts, and also provide new hypotheses about the specific contributions microbes make to ecosystem-scale metabolic processes.

# Introduction

Natural microbial communities have central roles in the biosphere, ranging from mediators of nutrient cycling to agents of human health and disease [@Falkowski2008a; @Blaser2016]. However, the majority of microbial species remain uncultivated, a feature that poses a significant challenge to our understanding of their physiology and metabolism. Recent advances in sequencing technology and bioinformatics have enabled assembly and analysis of reference genomes for a wide range of hitherto uncultured community members from diverse environments [@Sangwan2016] that can be used to reconstruct an organism's metabolism.

Common approaches to metabolic reconstruction involve the comprehensive reconstruction of an organism's metabolic pathways [@Lawson2017], or a targeted search for genes involved in processes of interest [@Anantharaman2016]. These reconstructions can then be analyzed manually or using computational approaches such as flux-balance analysis (FBA) [@Orth2010a]. However, FBA-based approaches require a comprehensive understanding of an organism's growth requirements and biomass composition, information which is often unavailable for uncultivated microorganisms. An alternative approach is to compute an organism's _seed set_, the set of compounds that the organism cannot synthesize on its own and must exogenously acquire from its environment (e.g., its growth requirements) [@Borenstein2008]. These compounds may represent both _auxotrophies_, essential metabolites for which biosynthetic routes are missing, and _nutrients_, compounds for which degradation but not synthesis routes are present in the genome. The _seed set framework_ offers potential advantages over other reconstruction-based approaches, as identification of seed compounds facilitates a focused analysis by identifying those compounds through which an organism interacts with its environment.

In the present study, we present a computational pipeline to predict seed compounds using metabolic network reconstructions generated from KBase [@Arkin2016]. We apply this pipeline to a collection of 36 metagenome-assembled genomes (MAGs) and single-cell genomes (SAGs) from the abundant and ubiquitous freshwater Actinobacterial lineage acI, which is thought to have a central role in nutrient cycling in diverse freshwater systems [@Zwart1998; @Glockner2000; @Zwart2002; @Newton2006; @Wu2006; @Newton2007; @Wu2007; @DeWever2008; @Humbert2009; @Ghai2012]. The seed compounds predicted by our analysis are in agreement with previous experimental and genomic observations [@Buck2009; @Salcher2010; @Martinez-Garcia2012; @Garcia2013; @Salcher2013; @Ghai2014; @Ghylin2014; @Tsementzi2014; @Garcia2015], confirming the ability of our method to predict an organism's auxotrophies and nutrient sources.

In particular, we find that members of the acI lineage are auxotrophic for essential vitamins and amino acids, and may consume as nutrients a wide array of N-containing compounds (including ammonium, branched-chain amino acids, polyamines, and di- and oligo-peptides) as well as mono-,  poly-, and oligo-saccharides. To complement these predictions, and to understand which pathways dominate active metabolism of acI in its natural environment, we conducted an _in situ_ metatranscriptomic analysis of gene expression in the acI lineage. This analysis revealed that the acI express a diverse array of transporters for auxotrophies, nutrients, and other compounds that may contribute to their observed dominance and widespread distribution in a variety of aquatic systems.

# Results

## Phylogenetic Affiliation of acI Genomes

We identified 17 SAGs and 19 MAGs from members of the acI lineage (Supplementary Table S1) in a larger set of reference genomes derived from our long-term study sites. A phylogenetic tree of these genomes built using a concatenated alignment of single-copy marker genes is shown in Figure 1. Previous phylogenetic analyses using 16S rRNA gene sequences showed that the acI lineage can be grouped into three distinct monophyletic clades (acI-A, acI-B, and acI-C) and thirteen so-called "tribes" [@Newton2011a]. In this study, the phylogenetic tree also identified three monophyletic branches, enabling MAGs to be classified to clade and tribe level based on the taxonomy of SAGs within each branch (as determined by 16S rRNA gene sequences, either PCR amplified or assembled from the single cell). Note that three MAGs formed a monophyletic group separate from clades acI-A and acI-B; we assume these genomes belong to clade acI-C as no other acI clades have been identified to date.

## Estimated Completeness of Tribe- and Clade-Level Composite Genomes

We constructed composite genomes from multiple SAGs and/or MAGs, to partially alleviate limitations presented by incomplete genomes. To do this, we first estimated the completeness of tribe- and clade-level composite genomes using CheckM [@Parks2015], which uses lineage-specific marker genes organized into collocated sets. This allowed us to determine the finest level of taxonomic resolution at which we could confidently compute seed compounds, using genome completeness as a proxy for metabolic reaction network completeness (Figure S2). We deemed genomes to be nearly complete if they contained 95% of the lineage-specific marker genes. With the exception of tribe acI-B1, tribe-level composite genomes are estimated to be incomplete (Figure S2A). At the clade level, clades acI-A and acI-B are estimated to be nearly complete, while the acI-C composite genome remains incomplete, as it only contains 75% of the 204 marker genes (Figure S2B). As a result, seed compounds were calculated for composite clade-level genomes, with the understanding that some true seed compounds for the acI-C clade will not be predicted.

## Computation and Evaluation of Potential Seed Compounds

Metabolic network reconstructions for each genome were built using KBase. Composite metabolic network graphs were then constructed for each tribe and clade by merging metabolic network reconstructions of individual genomes. Seed compounds for each clade were then computed from that clade's composite metabolic network graph using a custom implementation of the seed set framework (Figure 2). A total of 125 unique seed compounds were identified across the three clades (Table S2).

Because KBase is an automated annotation pipeline, the predicted set of seed compounds is likely to contain inaccuracies (e.g., due to missing or incorrect annotations). As a result, we screened the set of predicted seed compounds to identify those that represented biologically plausible auxotrophies and nutrients, and manually curated this subset to obtain a final set of auxotrophies and nutrient sources. Of 125 unique compounds, 31 (79%) were retained in the final set of proposed auxotrophies and nutrients. Tables S3 and S4 contain this final set of compounds for clades acI-A, acI-B, and acI-C, and Figure 3 shows the auxotrophies and nutrients these compounds represent.

## Making Sense of Seed Compounds via Protein Clustering and Metatranscriptomic Mapping

For seed compounds representing nutrient sources, genes associated with the consumption of these compounds should be expressed. To test this, we collected and sequenced four metatranscriptome samples from Lake Mendota (Dane County, WI, USA). However, because seed compounds were computed from each clade’s composite metabolic network graph, genes associated with the consumption of seed compounds may be present in multiple genomes within the clade. To facilitate the linkage of metatranscriptome measurements to seed compounds, we used OrthoMCL [@Li2003] to identify COGs in the set of acI genomes, merged metatranscriptome reads from all four samples, and mapped the reads to clusters of orthologous groups (COGs) within each clade.

Sequencing of cDNA from all four rRNA-depleted metatranscriptome samples yielded approximately 160 million paired-end reads. After merging, filtering, and further _in-silico_ rRNA removal, approximately 81 million, or 51% of the reads remained (Table S5). We then used BBMap (https://sourceforge.net/projects/bbmap/) to map metatranscriptome reads to our reference genome collection. After mapping the metatranscriptomes to our acI genomes, we calculated the average coverage of each genome in our reference collection. Within each clade, the most abundant genome was detected with at least 16-fold coverage (Table S6).

Finally, we calculated gene expression for each COG on a Reads Per Kilobase Million (RPKM) basis (Figure 2). OrthoMCL identified a total of 5013 protein clusters across the three clades (Table S7) with an average confidence of 84% in annotation for COGs containing more than one gene. The COGs were unequally distributed across the three clades, with clade acI-A genomes containing 3175 COGs (63%), clade acI-B genomes containing 3459 COGs (69%), and clade acI-C genomes containing 1365 COGs (27%). Of these, 525 COGs were expressed in clade acI-A, 661 in clade acI-B, and 813 in clade acI-C (Table S8). Among expressed genes, the median log2 RPKM value was 31.1 in clade acI-A, 32.0 in clade acI-B, and 69.4 in clade acI-C. Due to differing RPKM values in each clade, we report gene expression values for each clade as relative to the median for that clade.

## Auxotrophies and Nutrient Sources of the acI Lineage

Seed set analysis yielded seven auxotrophies that could be readily mapped to ecophysiological attributes of the acI lineage (Figure 3A and Table S3). In all three clades, beta-alanine was identified as a seed compound, suggesting an auxotrophy for pantothenic acid (Vitamin B5), a precursor to coenzyme A formed from beta-alanine and pantoate (Table S9). In bacteria, beta-alanine is typically synthesized via aspartate decarboxylation, and we were unable to identify a candidate gene for this enzyme (aspartate 1-decarboxylase, E.C. 4.1.1.11) in any acI genome. Pyridoxine 5'-phosphate and 5'-pyridoxamine phosphate (forms of the enzyme cofactor pyridoxal 5'-phosphate, Vitamin B6) were also predicted to be seed compounds, and numerous enzymes in the biosynthesis of these compounds were not found in the genomes (Table S9).

Clades within the acI lineage also exhibited distinct auxotrophies. Clade acI-A was predicted to be auxotrophic for the cofactor tetrahydrofolate (THF or Vitamin B9), and numerous enzymes for its biosynthesis were missing (Table S9). This cofactor plays an important role in the metabolism of amino acids and vitamins. In turn, clade acI-B was predicted to be auxotrophic for adenosylcobalamin (Vitamin B12), containing only four reactions from its biosynthetic pathway (Table S9). Finally, acI-C was predicted to be auxotrophic for the nucleotide uridine monophosphate (UMP, used as a monomer in RNA synthesis) and the amino acids lysine and homoserine. In all cases multiple enzymes for the biosynthesis of these compounds were not found in the acI-C genomes (Table S9).

A number of seed compounds were also predicted to be degraded by members of the acI lineage (Figure 3B and Table S3). Both clades acI-A and acI-B were predicted to use D-altronate and trans-4-hydroxy proline as nutrients, and acI-B was additionally predicted to use glycine betaine.

Finally, all three clades were predicted to use di-peptides and the sugar maltose as nutrients. Clades acI-A and acI-C were also predicted to consume the polysaccharides stachyose, manninotriose, and cellobiose. In all cases, these compounds were associated with reactions catalyzed by peptidases or glycoside hydrolases (Table S10 and S11), which may be capable of acting on compounds beyond the predicted seed compounds. Thus, we used these annotations to define nutrient sources, rather than using the predicted seed compounds themselves. Among these nutrient sources were di- and polypeptides, predicted to be released from both cytosolic- and membrane-bound aminopeptidases. As discussed below, we identified a number of transport proteins capable of transporting these released residues. In Lake Mendota, clades acI-B and acI-C expressed two aminopeptidases, one of which was expressed at upwards of 175% of the median gene expression levels (Table S10). Clade acI-A expressed a third aminopeptidase at a lower level of 40% the median gene expression level (Table S10).

All three clades were predicted to encode an alpha-glucosidase, which in Lake Mendota was only expressed in clades acI-B and acI-C, at upwards of 60% of the median gene expression level (Table S11). All three clades also encode a beta-glucosidase, but it was not expressed in our samples. Furthermore, all three clades encode an alpha-galactosidase and multiple maltodextrin glucosidases (which free maltose from maltotriose), but these were only expressed in clades acI-A and acI-C. The alpha-galactosidase had a log2 RPKM expression value of 1.5 times the median in clade acI-C, while the maltodextrin glucosidases were expressed at approximately 30% of the median (Table S11) in both clades acI-A and acI-C.

## Compounds Transported by the acI Lineage

Microbes may be capable of transporting compounds that are not strictly required for growth, and comparing such compounds to predicted seed compounds can provide additional information about an organism's ecology. Thus, we used the metabolic network reconstructions for the acI genomes to systematically characterize the transport capabilities of the acI lineage.

All acI clades encode for and expressed a diverse array of transporters (Figure 4, Tables S12 and S13, and the Supplemental Text S1). Consistent with the presence of peptidases, all clades contain numerous genes for the transport of peptides and amino acids, including putative oligopeptide and branched-chain amino acid transporters, as well as putative transporters for the polyamines spermidine and putrescine. All clades also contain a putative transporter for ammonium. The ammonium, branched-chain amino acid, and oligopeptide transporters had expression values above the median, with expression values for the substrate-binding protein (of the ATP-binding cassette (ABC) transporters) ranging from 1.7 to 411 times the median (Table S13). In contrast, while all clades expressed some genes from the polyamine transporters, only clade acI-B expressed the binding protein, at approximately 27.8 times the median (Table S13). Finally, clades acI-A and acI-B also contain a putative transporter for glycine betaine, which was only expressed in clade acI-A, at approximately 9.6 times the median (Table S13). However, we cannot rule out the possibility that the expression of these transporters changes with space and time, and that all three clades may express these enzymes under a different condition.

All clades also expressed transporters consistent with the presence of glycoside hydrolases, including transporters annotated as putative maltose, xylose, and ribose ABC-type transporters, which may indicate that acI bacteria are capable of transporting sugars, including both di- (maltose) and mono-saccharides (xylose and ribose). Of these, the putative maltose transporter was most highly expressed (but only in clades acI-A and acI-B), with expression values for the substrate-binding protein ranging in excess of 40 times the median (Table S13).

Representatives from the acI lineage also encode and expressed a number of transporters that do not have corresponding seed compounds, including potential nucleobase and purine/pyrimidine transporters (annotated as a uracil and a xanthine/uracil/thiamine/ascorbate family permease, respectively). Both of these are expressed in all three clades, with expression values ranging from 4.7 to 46 times the median (Table S13). Clades acI-A and acI-B also contain a second potential purine/pyrimidine transporter (annotated as a cytosine/purine/uracil/thiamine/allantoin family permease), which was only expressed in clade acI-B (Table S13). These transporters may be responsible for the uptake of the seed compounds UMP (a pyrimidine derivative) and Vitamin B1 (also known as thiamine). In addition, clade acI-A contains but did not express a putative transporter for cobalamin (Vitamin B12), and both clades acI-A and acI-B contain but did not express transporters for thiamin (Vitamin B1) and biotin (Vitamin B7) (Table S13).

Finally, all three clades expressed actinorhodopsin, a light-sensitive protein that is expected to function as a proton efflux pump [@Sharma2008]. In all clades, actinorhodopsin was among the top ten most highly-expressed genes (Table S7), with expression values in excess of 84 times the median in all three clades (Table S7). Given that many of the transport proteins are ABC transporters, we speculate that actinorhodopsin may facilitate maintenance of the proton gradient necessary for ATP synthesis. Coupled with high expression levels of diverse transporters, this result strongly suggests that acI functions as a photoheterotroph. However, it remains to be seen if this behavior is a general feature of acI physiology or restricted to the specific conditions of the lake and our sampling period.

# Discussion

This study uses high-throughput metabolic network reconstruction and the seed set framework to predict auxotrophies and nutrient sources of uncultivated microorganisms from incomplete genome sequences. The computational approach easily scales to hundreds of metabolic reconstructions, and enables a targeted analysis by identifying those compounds through which an organism interacts with its environment. However, predicted seed compounds are sensitive to the metabolic network structure, and analyzing the results requires significant manual curation of the metabolic reconstruction and accurate interpretation of the underlying gene annotations. As a consequence, the seed set framework as not as high-throughput as initially envisioned, but is nevertheless suitable for analysis of microorganisms with high-quality metabolic network reconstructions.

Our predictions of substrate use capabilities of the acI lineage are largely congruent with previous genome-centric studies based on smaller but manually curated genome collections [@Garcia2013; @Ghylin2014; @Garcia2015], indicating that the use of automatic metabolic network reconstructions yields similar predictions to comparatively more manual metabolic reconstruction efforts, while being both more high-throughput and focused on an organism's substrate utilization capabilities. In particular, this study predicts that the consumption of N-rich compounds is a universal feature of the acI lineage, with all three clades predicted to consume ammonium, branched-chain amino acids, polyamines, and di- and oligopeptides. These findings agree with MAR-FISH studies that confirm the ability of acI bacteria to consume a variety of amino acids [@Salcher2010; @Salcher2013]. Furthermore, the presence of alpha- and beta-glucosidases are consistent with observations that acI bacteria consume glucose [@Buck2009; @Salcher2013], even though no obvious glucose transport system was found in the genomes. Because transport proteins are often capable of acting on multiple substrates, one of the putative sugar transporters may be responsible for glucose uptake activity.

However, our approach failed to recapitulate other genomic and experimental observations, including the uptake of N-acetylglucosamine (NAG) [@Beier2011; @Eckert2012; @Eckert2013], the deoxynucleoside thymidine [@Perez2010; @Salcher2013], and acetate [@Buck2009], and the potential to hydrolyze the cyanobacterial peptide cyanophycin via the enzyme cyanophycinase [@Garcia2013; @Ghylin2014]. Inspection of these discrepancies reveals some important limitations of the seed set framework and automatic metabolic reconstructions. First, the seed set framework only identifies compounds that the metabolic network must obtain from its environment, and will fail to identify compounds that the organism can acquire from its environment but can also synthesize itself. Thymidine and acetate fall into this category. Second, automatic metabolic network reconstructions may not fully capture an organism's metabolic network (e.g., due to missing or incorrect genome annotations). Manual inspection of the previously-identified cyanophycinase gene revealed that KBase annotated this putative enzyme as a hypothetical protein. As biochemical characterization of hypothetical proteins and automatic gene and protein annotation are active areas of research, we anticipate that advances in these fields will continue to improve the accuracy of automatic metabolic network reconstructions.

This study also suggests that auxotrophies for some vitamins may be universal features of the acI lineage, as we predict all clades to be auxotrophic for pantothenic acid and pyridoxal 5'-phosphate (Vitamins B5 and B6). We also predict new auxotrophies within the acI lineage, including THF (clade acI-A), adenosylcobalamin (Vitamin B12, clade acI-B), and lysine, homoserine, and UMP (clade acI-C).  However, with the exception of adenosylcobalamin, we did not identify transporters for any of these compounds. This negative result may reflect our limited knowledge of transport proteins [@Saier2014]: transporters for these compounds may yet be present in the genomes, or one or more of the predicted transporters may act on these compounds. Furthermore, because the acI-C composite genome was estimated to be around 75% complete, we cannot rule out the possibility that the missing genes might be found in this clade when additional genomes are recovered. Nonetheless, these results provide additional support to the hypothesis that distributed metabolic pathways and metabolic complementarity may be common features of freshwater bacterial communities [@Garcia2016; @Garcia2017].

Combined, these results suggest that acI are photoheterotrophs, making a living on a diverse array of N-rich compounds, saccharides, and light. The acI lineage does not appear to be metabolically self-sufficient, and may participate in the turnover of high molecular weight dissolved organic compounds, such as starch, glycogen, and cellulose. Metatranscriptomic analysis showed that transport proteins were among the most highly expressed in the acI genomes, and the expression of multiple putative amino acid transporters may facilitate uptake of these labile compounds. We also observed differences in the relative expression of these transporters, which may point to clade-specific differences in the affinity for these substrates. Finally, the actinorhodopsin protein was highly expressed, and may facilitate synthesis of the ATP needed to drive acI's many ABC-type transporters.

Finally, the fragmented and incomplete nature of SAGs and MAGs required us to construct composite genomes for individual acI clades by leveraging multiple genomes from closely related populations. Such an approach limits the resolution of predictions, as we cannot make predictions at the level of tribes, smaller populations, or individual cells. Thus, metabolic diversification at these taxonomic levels will be missed. Constructing composite genomes may also overestimate the metabolic capabilities of a clade or group: for example, if a complete pathway is present in a clade but distributed among different tribes, the clade will only be able to carry out the entire pathway _in situ_ if all tribes are present in close enough proximity to exchange pathway intermediates. Nonetheless, the seed set approach provides a framework that can be used to generate new hypotheses about the substrates used by members of a defined phylogenetic group, provided multiple closely related genomes are available. As metagenomic assembly and binning techniques and single cell sequencing methods improve and complete genomes become available, we anticipate our approach being applied to individual microbial genomes.

# Materials and Methods

## A Freshwater Reference Genome Collection

This study relies on an extensive collection of freshwater bacterial genomes, containing MAGs obtained from two metagenomic time-series from two Wisconsin lakes [@Garcia2015; @Bendall2016], as well as SAGs from three lakes in the United States [@Martinez-Garcia2012]. Additional information about this genome collection can be found in the Supplemental Text S1.

## Metatranscriptome Sampling and Sequencing

This study used four metatranscriptomes obtained as part of a larger study of gene expression in freshwater microbial communities. Additional information about these samples can be found in the Supplemental Text S1. All protocols and scripts for sample collection, RNA extraction, rRNA depletion, sequencing, and bioinformatic analysis can be found on Github ([https://github.com/McMahonLab/OMD-TOIL](https://github.com/McMahonLab/OMD-TOIL), [DOI:10.5281/zenodo.839851](https://doi.org/10.5281/zenodo.839850)). Metadata for the four samples used in this study can be found in Table S6, and the raw RNA sequences can be found on the National Center for Biotechnology Information's Sequence Read Archive (SRA) under BioProject PRJNA362825.

## Identification of acI SAGs and Actinobacterial MAGs

The acI were previously phylogenetically divided into three clades (acI-A, acI-B, and acI-C) and thirteen tribes on the basis of their 16S rRNA gene sequences [@Newton2011a]. The acI SAGs were identified within a previously-published genome collection [@Martinez-Garcia2012] and classified to the tribe level using partial 16S rRNA genes and a reference taxonomy for freshwater bacteria, as described in the Supplemental Text S1. Actinobacterial MAGs were identified within two metagenomic time-series [@Garcia2015; @Bendall2016] using taxonomic assignments from a subset of conserved marker genes, as described in the Supplemental Text S1. Phylogenetic analysis of acI SAGs and Actinobacterial MAGs was performed using a concatenated alignment of single-copy marker genes obtained via Phylosift [@Darling2014]. Maximum likelihood trees were generated using RAxML [@Stamatakis2014] using the automatic protein model assignment option (PROTGAMMAAUTO) and 100 bootstraps.

## Genome Annotation, Metabolic Network Reconstruction, and Computation and Evaluation of Seed Compounds

In the seed set framework, an organism's metabolism is represented via a metabolic network graph, in which nodes denote compounds and edges denote enzymatically-encoded biochemical reactions linking substrates and products [@Jeong2000]. Allowable biochemical transformations can be identified by drawing paths along the network, in which a sequence of edges connects a sequence of distinct vertices. In our implementation of the seed set framework, metabolic network graphs were generated as follows.

Genome annotations were performed and metabolic network reconstructions were built using KBase. Contigs for each genome were uploaded to KBase and annotated using the "Annotate Microbial Contigs" method with default options, which uses components of the RAST toolkit for genome annotation [@Brettin2015; @Overbeek2014]. Metabolic network reconstructions were obtained using the "Build Metabolic Model" app with default parameters, which relies on the Model SEED framework [@Henry2010a] to build a draft metabolic model. No gap-filling was performed, to ensure that the reconstructions only contained reactions with genomic evidence. These reconstructions were then pruned (currency metabolites and highly-connected compounds) and converted to metabolic network graphs (Figure S3 and Supplemental Text S1). Many of the individual acI genomes are incomplete. Therefore, composite metabolic network graphs were constructed for each tribe and clade, to increase the accuracy of seed identification by means of a more complete metabolic network (Figure S4 and Supplemental Text S1).

Formally, the seed set of the network is defined as the minimal set of compounds that cannot be synthesized from other compounds in the network, and whose presence enables the synthesis of all other compounds in the network [@Borenstein2008]. Seed compounds for each composite metabolic network graph were calculated using a new Python implementation of the seed set framework [@Borenstein2008] (Figure S5 and the Supplemental Text S1). Because seed compounds are computed from a metabolic network, it is important to manually evaluate all predicted seed compounds to identify those that may be biologically meaningful, and do not arise from errors in the metabolic network reconstruction. Compounds involved in fatty acid and phospholipid biosynthesis pathways were removed during curation, as these pathways are often organism-specific and unlikely to be properly annotated by automatic metabolic reconstruction pipelines. Seed compounds related to currency metabolites (compounds used to carry electrons and functional groups) were also removed, as reactions for the synthesis of these compounds may have been removed during network pruning.

The Supplemental Text S1 contains a series of brief vignettes explaining why select compounds were discarded based on the afore-mentioned considerations, and provides examples of additional curation efforts applied to biologically plausible compounds. For a plausible auxotrophy, we screened the genomes for the canonical biosynthetic pathway(s) for that compound, and retained those compounds for which the biosynthetic pathway was incomplete. For a plausible nutrient source, we screened the genomes for the canonical degradation pathway(s) for that compound, and retained those compounds for which the degradation pathway was complete.

All computational steps were implemented using Python scripts, freely available as part of the reverseEcology Python package developed for this project ([https://pypi.python.org/pypi/reverseEcology/](https://pypi.python.org/pypi/reverseEcology/), DOI:[10.5281/zenodo.839856](https://doi.org/10.5281/zenodo.839856)).

## Identification of Transported Compounds

For each genome, we identified all transport reactions present in its metabolic network reconstruction. Gene-protein-reaction associations (GPRs) for these reactions were manually curated to remove unannotated proteins, group genes into operons (if applicable), and to identify missing subunits for multi-subunit transporters. These genes were then mapped to their corresponding COGs, and grouped accordingly. Finally, the most common annotation for each COG was used to identify likely substrates for each of these groups. Only transporters with >50% confidence in the substrate-binding subunit were retained. Because identification and annotation of transport proteins is an active area of research [@Saier2014], substrates for each transporter are described as putative and acting on molecular classes (e.g., saccharide, amino acid) instead of specific compounds, to better reflect the promiscuity of transport proteins and the ambiguity of their annotation.

## Protein Clustering, Metatranscriptomic Mapping, and Clade-Level Gene Expression

OrthoMCL [@Li2003] was used to identify clusters of orthologous groups (COGs) in the set of acI genomes. Both OrthoMCL and BLAST were run using default options [@Fischer2011]. Annotations were assigned to protein clusters by choosing the most common annotation among all genes assigned to the respective cluster and a confidence score assigned to each COG (fraction of genes having the most common annotation). Trimmed and merged metatranscriptomic reads from each of the four biological samples were then pooled and mapped to a single reference fasta file containing all acI genomes using BBMap with the _ambig=random_ and _minid=0.95_ options. The 95% identity cutoff was chosen as this represents a well-established criterion for identifying microbial species using average nucleotide identity (ANI) [@Konstantinidis2005a], while combining the _ambig_ option with competitive mapping using pooled acI genomes as the reference ensures that reads map only to a single genome. These results were then used to compute the expression of each COG in each clade.

Next, HTSeq-Count [@Anders2014] was run using the _intersection_strict_ option to count the total number of reads that map to each gene in our acI genome collection. After mapping, the list of counts was filtered to remove those genes that did not recruit at least ten reads. Using the COGs identified by OrthoMCL, the genes that correspond to each COG were then identified.

Within each clade, gene expression for each COG was computed on a Reads Per Kilobase Million (RPKM) basis [@Mortazavi2008], while also accounting for different gene lengths within a COG and numbers of mapped reads for each genome within a clade. That is, the RPKM value for a single COG represents the sum of RPKM values for each gene within that COG, normalized to the appropriate gene length and total number of mapped reads. RPKM counts were then normalized to the median level of gene expression within that clade. Finally, the expression data (mapping of transcript reads to genes) were visualized to ensure RPKM calculations were based on continuous transcription of each gene.

## Availability of Data and Materials

All genomic and metatranscriptomic sequences are available through IMG and NCBI, respectively. A reproducible version of this manuscript is available at [https://github.com/joshamilton/Hamilton_acI_2017](https://github.com/joshamilton/Hamilton_acI_2017) (DOI:[10.5281/zenodo.839858](https://doi.org/10.5281/zenodo.839858)).

# Acknowledgements

We thank past and present members of the McMahon lab for collecting water samples for DNA and RNA sequencing and Frank Aylward for guidance on metatranscriptomic analysis. DNA sequencing was supported through the JGI Community Science Program. The work conducted by the JGI, a DOE Office of Science User Facility, is supported by the Office of Science of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231. This material is based upon work that is supported by the National Institute of Food and Agriculture, U.S. Department of Agriculture, under award number 2016-67012-24709 to JJH and WIS01789 to KDM. KDM also acknowledges funding from the United States National Science Foundation (NSF) Microbial Observatories program (MCB-0702395), the NSF Long Term Ecological Research program (NTL-LTER DEB-1440297), an NSF INSPIRE award (DEB-1344254), and the University of Wisconsin System. KDM and KTF acknowledge National Oceanic and Atmospheric Administration (NOAA) grant #NA10OAR4170070, Wisconsin Sea Grant College Program Project #HCE-25, through NOAA’S National Sea Grant College Program, U.S. Department of Commerce. The funders had no role in study design, data collection and interpretation, or the decision to submit the work for publication.

# Conflict of Interest

The authors declare no conflict of interest.

# Figure Legends

&nbsp;

## Figure 1 (figures/fig1-tree/Figure1.pdf)

&nbsp;

Phylogenetic placement of the genomes used in this study within the acI lineage. The tree was built using RAxML [@Stamatakis2014] from a concatenated alignment of protein sequences from 37 single-copy marker genes [@Darling2014]. The order Actinomycetales forms the outgroup. Vertical black bars indicate groups of genomes belonging to defined tribes/clades within the acI lineage, as determined using 16S rRNA gene sequences (for SAGs and bin FNEF8-2 bin_7 acI-B only) and a defined taxonomy [@Newton2011a]. SAGs are indicated with italic text. Supplementary Figure S1 shows the position of the acI lineage relative to other orders within the class Actinobacteria.

&nbsp;

## Figure 2 (figures/fig2-workflow/Figure2.pdf)

&nbsp;

Overview of the seed set framework and metatranscriptomic mapping, using three genomes from the acI-C clade as an example. __(A)__ Metabolic network graphs are created for each genome belonging to clade acI-C. In these graphs, metabolites are represented as nodes (circles) and reactions by arcs (arrows). Grey nodes and edges indicate components of the composite graph missing from that genome graph. Additional information on this step of the workflow is available in Figure S2. __(B)__ A composite network graph is created for each clade by joining graphs for all genomes from that clade, and seed compounds (red) are computed for the composite graph. Additional information on this step of the workflow is available in Figures S3, S4, and S5. __(Inset)__ Three seed compounds which indicate an auxotrophy for L-homoserine, a methionine precursor. __(C)__ Metatranscriptomic reads are mapped to each individual genome using BBMap. Orthologous gene clusters are identified using OrthoMCL [@Li2003]. For each cluster, unique reads which map to any gene within that cluster are counted using HTSeq [@Anders2014]. The relative gene expression is computed using RPKM [@Mortazavi2008].

&nbsp;

## Figure 3 (figures/fig3-seeds/Figure3.pdf)

&nbsp;

Seed compounds of members of the acI lineage. __(A)__ Auxotrophies and nutrient sources, not including peptides and glycosides. __(B)__ Peptides and glycosides. These compounds represent those inferred from genome annotations, rather than the seed compounds themselves. In panel (B), the intensity of the color indicates the log2 RPKM of the encoding gene cluster. For compounds acted upon by multiple gene clusters, the percentile of the most highly-expressed cluster was chosen.

&nbsp;

## Figure 4 (figures/fig4-transporters/Figure4.pdf)

&nbsp;

Transporters that are actively expressed by members of the acI lineage, as inferred from consensus annotations of genes associated with transport reactions present in metabolic network reconstructions. The intensity of the color indicates the log2 fold-change relative to the median of the encoding gene cluster. For multi-subunit transporters, the RPKM of the substrate-binding subunit was chosen (Table S13). For some transporters, consensus annotations have been replaced with broad metabolite classes. Such metabolite classes are indicated with superscripts, and the original annotations are as follows: 1 - spermidine, putrescine; 2 - maltose ; 3 - xylose; 4 - ribose; 5 - uracil; 6 - cytosine / purine / uracil / thiamine / allantoin; 7 - xanthine / uracil / thiamine / ascorbate.

&nbsp;

# Supplementary Material Legends

&nbsp;

## Supplementary Figure 1 (figures/fig1-tree/FigureS1.pdf)

&nbsp;

Phylogenetic placement of the genomes used in this study within the acI lineage, relative to other sequenced actinobacterial genomes in the class Actinobacteria [@Gao2012] (Table S19). The tree was built using RAxML [@Stamatakis2014] from a concatenated alignment of protein sequences from 37 single-copy marker genes [@Darling2014]. The class Acidimicrobiia forms the outgroup. Vertical black bars indicate groups of genomes belonging to defined tribes/clades within the acI lineage, as determined using 16S rRNA gene sequences (for SAGs and bin FNEF8-2 bin_7 acI-B only) and a defined taxonomy [@Newton2011a]. SAGs are indicated with italic text.

&nbsp;

## Supplementary Figure 2 (figures/figS-sampling/FigureS1.pdf)

&nbsp;

Mean estimated completeness of tribe-level (clade-level) population genomes as a function of the number of sampled genomes. For each tribe (clade), genomes were randomly sampled (with replacement) from the set of all genomes belonging to that tribe (clade). Completeness was estimated using 204 single-copy marker genes from the phylum Actinobacteria [@Parks2015]. Error bars represent the 95% confidence interval estimated from 1000 iterations.

&nbsp;

## Supplementary Figure 3 (figures/fig2-workflow/FigureS3.pdf)

&nbsp;

Converting an unannotated genome to a metabolic network graph, for a simplified genome containing only glycolysis. __(A)__ Microbial contigs are annotated using KBase, and a metabolic network reconstruction is built from the annotations. The reconstruction provides links between protein-encoding genes in the genome and the enzymatic reactions catalyzed by those proteins. __(B)__ The metabolic network reconstruction represents metabolism as a hypergraph, in which metabolites are represented as nodes and reactions as hyperedges. In this representation, an edge can connect more than two nodes. For example, a single hyperedge (denoted by a heavy black line) connects the metabolites glucose and ATP to glucose-6P, ADP, and Pi. For clarity, protons are not shown. __(C)__ However, the algorithm used by the seed set framework requires metabolism to be represented as a metabolic network graph, in which an edge can connect only two nodes. In this representation, a reaction is represented by a set of edges connecting all substrates to all products. For example, the heavy hyperedge in (B) is now denoted by six separate edges connecting glucose to ADP, glucose to Pi, glucose to glucose-6P, ATP to ADP, ATP to Pi, and ATP to glucose-6P (again denoted by heavy black lines). Of these, only one (glucose to glucose-6P) is biologically meaningful. The dotted line surrounds the currency metabolites. __(D)__ The metabolic network graph is then pruned, a process which removes all currency metabolites and any edges in which those metabolites participate. Of the six heavy edges in (C), only the biologically meaningful one is retained, connecting glucose to glucose-6P (again denoted by a heavy black line). The images in (B) and (C) are modified from [@Ma2003].

&nbsp;

## Supplementary Figure 4 (figures/fig2-workflow/FigureS4.pdf)

&nbsp;

Construction of composite metabolic network graph for clade acI-C. Beginning with metabolic network graphs for genomes Actinobacterium_10 and ME00885, nodes and edges unique to ME00885 are identified (in blue). These nodes and edges are added to the Actinobacterium_10 graph, giving the composite metabolic network graph for these two genomes (Actinobacterium_10 + ME00885). Then, this graph is compared to the graph for ME03864, and nodes and edges unique to ME03864 are identified (in blue). These nodes and edges are added to the Actinobacterium_10 + ME00885 metabolic network graph, giving the composite metabolic network graph for clade acI-C.

&nbsp;

## Supplementary Figure 5 (figures/fig2-workflow/FigureS5.pdf)

&nbsp;

Identifying seed compounds in metabolic networks, using the same metabolic network as in Supplementary Figure S1. __(A)__ To identify seed compounds, the metabolic network graph is first decomposed into its strongly connected components (SCCs), sets of nodes such that each node in the set is reachable from every other node. Here, each set of circled nodes corresponds to a unique SCC. __(B)__ SCC decomposition enables seed sets to be identified from source components (components with no incoming edges) on the condensation of the original graph. In the condensation of the original graph shown here, each node corresponds to a unique SCC. This network has a single seed set, SCC_1, enclosed in a dotted circle. __(C)__ Seed compounds can be found from the mapping between SCCs and their constituent metabolites. In this example, glucose is the sole seed compound. While this particular result is probably intuitive, real metabolic networks are considerably more complex.  Note: The visual representations shown here are intended to illustrate the metabolic network reconstruction process, and are not indicative of the data structures used by our pipeline.

&nbsp;

## Supplementary Figure 6 (figures/fig2-workflow/FigureS6.pdf)

&nbsp;

Complete composite metabolic network graph for clade acI-C, showing disconnected components (gray) and the single largest component (green and black). Disconnected components are dropped prior to computing the network's seed sets because these groups of nodes are not connected to the bulk of the network. Within the single largest component, the giant strong component contains a substantial fraction of the compounds (green nodes), giving rise to a bow-tie structure in the metabolic network graph.

&nbsp;

## Supplementary Tables (manuscript/Hamilton_acI_2017_Tables.xlsx)

&nbsp;

Supplementary Tables S1 to S19.

# References
