---
title: High-Throughput Metabolic Network Analysis and Metatranscriptomics of a Cosmopolitan and Streamlined Freshwater Microbial Lineage
author: Joshua J. Hamilton^1\*^, Sarahi L. Garcia^2, Brittany S. Brown^1, Ben O. Oyserman^3, Francisco Moya-Flores^3, Stefan Bertilsson^2,4, Rex R. Malmstrom^5, Katrina T. Forest^1, Katherine D. McMahon^1,3
abstract: \* Correspondence&#58; Joshua J. Hamilton, jjhamilton2@wisc.edu
date: ^1^ Department of Bacteriology, University of Wisconsin-Madison, Madison, WI, USA; ^2^ Department of Ecology and Genetics, Uppsala University, Uppsala, Sweden; ^3^ Department of Civil and Environmental Engineering, University of Wisconsin-Madison, Madison, WI, USA; ^4^ Science for Life Laboratory, Uppsala University, Uppsala, Sweden; ^5^ United States Department of Energy Joint Genome Institute, Walnut Creek, CA, USA
---

# Abstract

An explosion in the number of available genome sequences obtained through metagenomics and single-cell genomics has enabled a new view of the diversity of microbial life, yet we know surprisingly little about how microbes interact with each other or their environment. In fact, the majority of microbial species remain uncultivated, while our perception of their ecological niches is based on reconstruction of their metabolic potential. In this work, we demonstrate how the "seed set framework", which computes the set of compounds that an organism must acquire from its environment [@Borenstein2008], enables high-throughput, computational analysis of metabolic reconstructions, while providing new insights into a microbe's metabolic capabilities, such as nutrient use and auxotrophies. We apply this framework to members of the ubiquitous freshwater Actinobacterial lineage acI, confirming and extending previous experimental and genomic observations implying that acI bacteria are heterotrophs reliant on peptides and saccharides. We also present the first metatranscriptomic study of the acI lineage, featuring high expression of transport proteins and the light-harvesting protein actinorhodopsin, and confirming predictions of nutrients and essential metabolites while providing additional support to the hypothesis that members of the acI are photoheterotrophs.

# Introduction

Natural microbial communities have central roles in the biosphere, ranging from mediators of nutrient cycling to agents of human health and disease [@Falkowski2008a; @Blaser2016]. However, the majority of microbial species remain uncultivated, a feature that poses a significant challenge to our understanding of their physiology and metabolism. Recent advances in sequencing technology and bioinformatics have enabled assembly and analysis of reference genomes for a wide range of hitherto uncultured community members from diverse environments [@Sangwan2016] that can be explored for inferring links between the genome content of an individual microbe and its metabolic traits, a concept referred to as "reverse ecology" [@Levy2012].

Reverse ecological analyses can be performed using metabolic network reconstructions [@Feist2009a; @Thiele2010c], which are structured summaries of the metabolic capabilities of an organism as defined by its enzymes and their coupled biochemical reactions. These reconstructions can then be analyzed using metabolic network graphs, mathematical objects in which biochemical reactions are represented as connections between substrates and products [@Levy2012]. One such graph-based, reverse ecology approach is to compute an organism's _seed set_, the set of compounds that the organism cannot synthesize on its own and must exogenously acquire from its environment [@Borenstein2008]. These compounds may represent both _auxotrophies_ for essential metabolites for which biosynthetic routes are missing, and _nutrients_ for which degradation but not synthesis routes are present in the genome. This _seed set framework_ offers potential advantages over other reconstruction-based approaches, as 1) metabolic network graphs can be rapidly analyzed computationally, 2) a network-centric approach makes no _a priori_ assumptions about which metabolic pathways may be important for an organism's niche, and 3) identification of seed compounds facilitates a focused analysis by identifying those compounds that an organism must obtain from its environment.

Freshwater lake microbiomes are ideal systems for applying the seed set framework, as long-term monitoring has revealed the ecology of dominant bacterial lineages [@Newton2011a], and reference genomes for these key lineages are now readily available [@Martinez-Garcia2012; @Garcia2013; @Ghai2014; @Ghylin2014; @Tsementzi2014; @Garcia2015; @Bendall2016]. Among the freshwater bacteria, uncultivated Actinobacteria of the acI lineage are among the most abundant [@Zwart1998; @Glockner2000; @Zwart2002]. The acI have been phylogenetically divided into three clades (acI-A, acI-B, and acI-C) and thirteen tribes on the basis of their 16S rRNA gene sequences [@Newton2011a]. The abundance and ubiquitous distribution of these free-living ultramicrobacteria suggests they have central roles in nutrient cycling in diverse freshwater systems [@Glockner2000; @Newton2006; @Wu2006b; @Newton2007; @Wu2007; @DeWever2008; @Humbert2009; @Ghai2012].

To identify the metabolic processes that these bacteria mediate, the metabolism of the acI lineage has been the focus of a number of recent studies. Surveys using fluorescent _in situ_ hybridization (FISH) and catalyzed reporter deposition (CARD) or microautoradiography (MAR) reveal that the acI are capable of consuming amino acids [@Salcher2010; @Salcher2013], glucose [@Buck2009; @Salcher2013], N-acetylglucosamine (NAG) [@Beier2011; @Eckert2012; @Eckert2013], the deoxynucleoside thymidine [@Perez2010; @Salcher2013], and acetate [@Buck2009]. Furthermore, metabolic reconstructions of single-cell genomes (SAGs) and metagenome-assembled genomes (MAGs) have further expanded the view of substrate uptake capabilities for members of clades acI-A and acI-B. These studies indicate members of these clades are capable of consuming a wide array of N-containing compounds, including ammonium, branched-chain amino acids, polyamines, di- and oligo-peptides, and cyanophycin [@Ghylin2014; @Garcia2015]. Members of these two clades also seem to be capable of consuming numerous mono-,  poly-, and oligo-saccharides [@Garcia2013; @Ghylin2014; @Garcia2015]. Finally, a recent study of a metagenome-assembled genome from clade acI-B predicted that some members of the clade are unable to synthesize a number of essential vitamins and amino acids [@Garcia2015].

In the present study, we present a computational pipeline to automate the calculation of an organism's substrate utilization capabilities using the seed set framework, thereby facilitating high-throughput analysis of genomic data. To do so, we developed a Python package to predict seed compounds, using metabolic network reconstructions generated from KBase [@Arkin2016]. We expand existing analyses of the acI lineage by applying the seed set framework to a reference genome collection of 36 freshwater acI genomes covering all three acI clades. The seed compounds predicted by our analysis are in agreement with previous experimental and genomic observations, confirming the ability of our method to predict an organism's auxotrophies and nutrient sources. To complement these predictions, and to understand which pathways dominate active metabolism of acI in its natural environment, we conducted the first _in situ_ metatranscriptomic analysis of gene expression in the acI lineage. Knowledge of seed compounds enhanced interpretation of the metatranscriptome results by facilitating a focused analysis of acI metabolism. Additional analyses show that the acI express a diverse array of transporters that we hypothesize may contribute to their observed dominance and widespread distribution in a variety of aquatic systems.

# Materials and Methods

## A Freshwater Reference Genome Collection

This study relies on an extensive collection of freshwater bacterial genomes, containing MAGs obtained from two metagenomic time-series from two Wisconsin lakes [@Bendall2016; @Garcia2016a], as well as SAGs from three lakes in the United States [@Martinez-Garcia2012]. Additional information about this genome collection can be found in the Supplemental Online Material.

## Metatranscriptome Sampling and Sequencing

This study used four metatranscriptomes obtained as part of a 24-hour sampling experiment designed to identify diel trends in freshwater microbial communities. Additional information about these samples can be found in the Supplemental Online Material. All protocols and scripts for sample collection, RNA extraction, rRNA depletion, sequencing, and bioinformatic analysis can be found on Github (https://github.com/McMahonLab/OMD-TOILv2, DOI:######). Metadata for the four samples used in this study can be found in Table S1, and the raw RNA sequences can be found on the National Center for Biotechnology Information (NCBI) website under BioProject PRJNA362825.

## Identification of acI SAGs and Actinobacterial MAGs

acI SAGs were identified within a previously-published genome collection [@Martinez-Garcia2012] and classified to the tribe level using partial 16S rRNA genes and a reference taxonomy for freshwater bacteria, as described in the Supplemental Online Material. Actinobacterial MAGs were identified within two metagenomic time-series [@Bendall2016; @Garcia2016a] using taxonomic assignments from a subset of conserved marker genes, as described in the Supplemental Online Material. Phylogenetic analysis of acI SAGs and Actinobacterial MAGs was performed using a concatenated alignment of single-copy marker genes obtained via Phylosift [@Darling2014]. Maximum likelihood trees were generated using RAxML [@Stamatakis2014] using the automatic protein model assignment option (PROTGAMMAAUTO) and 100 bootstraps.

## Genome Annotation, Metabolic Network Reconstruction, and Computation and Evaluation of Seed Compounds

In the seed set framework, an organism's metabolism is represented via a metabolic network graph, in which nodes denote compounds and edges denote enzymatically-encoded biochemical reactions linking substrates and products [@Jeong2000]. Allowable biochemical transformations can be identified by drawing paths along the network, in which a sequence of edges connects a sequence of distinct vertices. In our implementation of the seed set framework, metabolic network graphs were generated as follows.

Genome annotations were performed and metabolic network reconstructions were built using KBase. Contigs for each genome were uploaded to KBase and annotated using the "Annotate Microbial Contigs" method with default options, which uses components of the RAST toolkit for genome annotation [@Brettin2015; @Overbeek2014]. Metabolic network reconstructions were obtained using the "Build Metabolic Model" app with default parameters, which relies on the Model SEED framework [@Henry2010a] to build a draft metabolic model. These reconstructions were then pruned (currency metabolites and highly-connected compounds) and converted to metabolic network graphs (Figure S1 and Supplemental Online Material). Many of the individual acI genomes are incomplete (see Results). Therefore, composite metabolic network graphs were constructed for each tribe and clade, to increase the accuracy of seed identification by means of a more complete metabolic network (Figure S2 and Supplemental Online Material).

Formally, the seed set of the network is defined as the minimal set of compounds that cannot be synthesized from other compounds in the network, and whose presence enables the synthesis of all other compounds in the network [@Borenstein2008]. Seed compounds for each composite metabolic network graph were calculated using a new Python implementation of the seed set framework [@Borenstein2008] (Figure S3 and the Supplemental Online Material). Because seed compounds are computed from a metabolic network, it is important to manually evaluate all predicted seed compounds to identify those that may be biologically meaningful, and do not arise from errors in the metabolic network reconstruction. Examples of this process are given in the Supplemental Online Material.

All computational steps were implemented using Python scripts, freely available as part of the reverseEcology Python package developed for this project (https://pypi.python.org/pypi/reverseEcology/, DOI:######).

## Identification of Transported Compounds

For each genome, we identified all transport reactions present in its metabolic network reconstruction. Gene-protein-reaction associations (GPRs) for these reactions were manually curated to remove unannotated proteins, group genes into operons (if applicable), and to identify missing subunits for multi-subunit transporters. These genes were then mapped to their corresponding COGs, and grouped accordingly. Finally, the most common annotation for each COG was used to identify likely substrates for each of these groups.

## Protein Clustering, Metatranscriptomic Mapping, and Clade-Level Gene Expression

OrthoMCL [@Li2003] was used to identify clusters of orthologous groups (COGs) in the set of acI genomes. Both OrthoMCL and BLAST were run using default options [@Fischer2011]. Annotations were assigned to protein clusters by choosing the most common annotation among all genes assigned to the respective cluster. Trimmed and merged metatranscriptomic reads from each of the four biological samples were then mapped to a single reference fasta file containing all acI genomes using BBMap (https://sourceforge.net/projects/bbmap/) with the _ambig=random_ and _minid=0.95_ options. The 95% identity cutoff was chosen as this represents a well-established criterion for identifying microbial species using average nucleotide identity (ANI) [@Konstantinidis2005a], while combining the _ambig_ option with competitive mapping using pooled acI genomes as the reference ensures that reads map only to a single genome. These results were then used to compute the expression of each COG in each clade.

Next, HTSeq-Count [@Anders2014] was used to count the total number of reads that map to each gene in our acI genome collection. After mapping, the list of counts was filtered to remove those genes that did not recruit at least one read in all four samples. Using the COGs identified by OrthoMCL, the genes that correspond to each COG were then identified.

Within each clade, gene expression for each COG was computed on a Reads Per Kilobase Million (RPKM) basis [@Mortazavi2008], while also accounting for different gene lengths within a COG and numbers of mapped reads for each genome within a clade. That is, the RPKM value for a single COG represents the sum of RPKM values for each gene within that COG, normalized to the appropriate gene length and total number of mapped reads. RPKM counts were then averaged across the four metatranscriptomes and normalized to the median level of gene expression within that clade.

## Availability of Data and Materials

All genomic and metatranscriptomic sequences are available through IMG and NCBI, respectively. A reproducible version of this manuscript is available at https://github.com/joshamilton/Hamilton_acI_2016 (DOI:######).

# Results

## Phylogenetic Affiliation of acI Genomes

From a reference collection of freshwater bacterial genomes, we identified 17 SAGs and 19 MAGs from members of the acI lineage. A phylogenetic tree of these genomes is shown in Figure 1. Previous phylogenetic analysis using 16S rRNA gene sequences have revealed that the acI lineage can be grouped into three distinct monophyletic clades [@Newton2011a]. In this study, the phylogenetic tree built from 37 concatenated marker genes also identified three monophyletic branches, enabling MAGs to be classified as clade acI-A or acI-B based on the taxonomy of SAGs within each branch. Note that three MAGs formed a monophyletic group separate from clades acI-A and acI-B; we assume these genomes belong to clade acI-C as no other acI clades have been identified to date.

## Estimated Completeness of Tribe- and Clade-Level Composite Genomes

Metabolic network reconstructions created from acI SAGs and MAGs will likely be missing reactions, as the underlying genomes are incomplete (Table 1). Previous studies have examined the effect of genome incompleteness on the predicted seed set [@Borenstein2008]. Using the formal (mathematical) definition of a seed compound, this showed that the percentage of correct seed compounds (true positives) is approximately equal to the completeness of the reaction network, and the number of false positives is approximately equal to the incompleteness of the network. Thus, we constructed composite genomes at higher taxonomic levels (e.g., tribe and clade) to increase genome completeness for more accurate seed identification at that taxonomic level.

Using 204 conserved single-copy marker genes [@Parks2015], we estimated the completeness of tribe- and clade-level composite genomes to determine the finest level of taxonomic resolution at which we could confidently compute seed compounds, using genome completeness as a proxy for metabolic reaction network completeness (Figure S5). Because CheckM relies on lineage-specific marker genes, the completeness of genomes without representation in the CheckM database can be underestimated [@Garcia2015]. As a result, we deemed genomes to be complete if they contained 95% of the lineage-specific marker genes. With the exception of tribe acI-B1, tribe-level composite genomes are estimated to be incomplete (Figure S5A). At the clade level, clades acI-A and acI-B are estimated to be complete, while the acI-C composite genome remains incomplete, as it only contains 75% of the 204 marker genes (Figure S5B). As a result, seed compounds were calculated for composite clade-level genomes, with the understanding that some true seed compounds for the acI-C clade will not be predicted.

## Computation and Evaluation of Potential Seed Compounds

Seed compounds were computed for each clade, using the composite metabolic network graph for that clade (Figure 2, and Figures S1 to S3). A total of 125 unique seed compounds were identified across the three clades (Table S2). Additional details are available in the Supplemental Online Material.

Seed compounds were predicted using the results of an automated annotation pipeline, and as such are likely to contain inaccuracies (e.g., due to missing or incorrect annotations). As a result, we screened the set of predicted seed compounds to identify those that represented biologically plausible auxotrophies and nutrients, and manually curated this subset to obtain a final set of auxotrophies and nutrient sources. Compounds involved in fatty acid and phospholipid biosynthesis pathways were removed during curation, as these pathways are often organism-specific and unlikely to be properly annotated by automatic metabolic reconstruction pipelines. Seed compounds related to currency metabolites were also removed, as reactions for the synthesis of these compounds may have been removed during network pruning. Of 125 unique compounds, 39 (31%) passed this screening and were deemed biologically plausible.

The Supplemental Online Material contains a series of brief vignettes explaining why select compounds were discarded based on the afore-mentioned considerations, and provides examples of additional curation efforts applied to biologically plausible compounds. For a plausible auxotrophy, we screened the genomes for the canonical biosynthetic pathway(s) for that compound, and retained those compounds for which the biosynthetic pathway was incomplete. For a plausible nutrient source, we screened the genomes for the canonical degradation pathway(s) for that compound, and retained those compounds for which the degradation pathway was complete. Of the 39 compounds deemed biologically plausible auxotrophies and nutrients, 31 (79%) were retained in the final set of proposed auxotrophies and nutrients. Tables S6 and S7 contain this final set of compounds for clades acI-A, acI-B, and acI-C, and Figure 3 shows the auxotrophies and nutrients these compounds represent.

## Making Sense of Seed Compounds via Protein Clustering and Metatranscriptomic Mapping

With regards to seed compounds representing nutrient sources, genes associated with the consumption of these compounds should be expressed. However, because seed compounds were computed from each clade’s composite metabolic network graph, genes associated with the consumption of seed compounds may be present in multiple genomes within the clade. To facilitate the linkage of metatranscriptome measurements to seed compounds, we decided to map metatranscriptome samples to clusters of orthologous groups (COGs) within each clade. We used OrthoMCL [@Li2003] to identify COGs in the set of acI genomes, and counted each COG as present in a clade if that COG was present in at least one genome belonging to that clade. We then used BBMap to map metatranscriptome reads to our reference genome collection, and counted the unique reads which map to each Actinobacterial COG.

Sequencing of cDNA from all four rRNA-depleted metatranscriptome samples yielded approximately 160 billion paired-end reads. After merging, filtering, and further _in-silico_ rRNA removal, approximately 81 billion, or 51% of the reads remained (Table S1). OrthoMCL identified a total of 5013 protein clusters across the three clades (Table S3). The COGs were unequally distributed across the three clades, with clade acI-A genomes containing 3175 COGs (63%), clade acI-B genomes containing 3459 COGs (69%), and clade acI-C genomes containing 1365 COGs (27%). After mapping the metatranscriptomes to our acI genomes (Table S4), we identified 650 COGs expressed in clade acI-A, 785 in clade acI-B, and 849 in clade acI-C (Table S5). Among expressed genes, the median log2 average RPKM value was 10.3 in clade acI-A, 10.2 in clade acI-B, and 9.0 in clade acI-C. Thus, despite differential abundance of each clade within the lake, median gene expression within each clade was similar.

## Auxotrophies and Nutrient Sources of the acI Lineage

Seed set analysis yielded seven auxotrophies that could be readily mapped to ecophysiological attributes of the acI lineage (Figure 3a). In all three clades, beta-alanine was identified as a seed compound, suggesting an auxotrophy for pantothenic acid (Vitamin B5), a precursor to coenzyme A formed from beta-alanine and pantoate. In bacteria, beta-alanine is typically synthesized via aspartate decarboxylation, and we were unable to identify a candidate gene for this enzyme (aspartate 1-decarboxylase, E.C. 4.1.1.11) in any acI genome. Pyridoxine 5'-phosphate and 5'-pyridoxamine phosphate (forms of the enzyme cofactor pyridoxal 5'-phosphate, Vitamin B6) were also predicted to be seed compounds, and numerous enzymes in the biosynthesis of these compounds were not found in the genomes.

Clades within the acI lineage also exhibited distinct auxotrophies. Clade acI-A was predicted to be auxotrophic for the cofactor tetrahydrofolate (THF or Vitamin B9), and numerous enzymes for its biosynthesis were missing. This cofactor plays an important role in the metabolism of amino acids and vitamins. In turn, clade acI-B was predicted to be auxotrophic for adenosylcobalamin (Vitamin B12), containing only a single reaction from its biosynthetic pathway. Finally, acI-C was predicted to be auxotrophic for the nucleotide uridine monophosphate (UMP, used as a monomer in RNA synthesis) and the amino acids lysine and homoserine. In all cases multiple enzymes for the biosynthesis of these compounds were not found in the acI-C genomes. However, with the exception of adenosylcobalamin, we did not identify transporters for any of these compounds. Furthermore, because the acI-C composite genome was estimated to be around 75% complete, we cannot rule out the possibility that the missing genes might be found in when additional genomes are recovered.

A number of seed compounds were predicted to be nutrients, compounds which can be degraded by members of the acI lineage (Figure 3B). Both clades acI-A and acI-B were predicted to use D-altronate and trans-4-hydroxy proline as nutrients, and acI-B was additionally predicted to use glycine betaine. These compounds indicate that the acI may participate in the turnover of plant- and animal-derived organic material in freshwater systems: glycine betaine is an important osmolyte in plants [@Ashraf2007], D-altronate is produced during degradation of galacturonate, a component of plant pectin [@Mohnen2008], and trans-4-hydroxy-L-proline is a major component of animal collagen [@Eastoe1955].

Finally, all three clades were predicted to use di-peptides and the sugar maltose as nutrients. Clades acI-A and acI-C were also predicted to consume the polysaccharides stachyose, manninotriose, and cellobiose. In all cases, these compounds were associated with reactions catalyzed by peptidases or glycoside hydrolases (Table S8 and S9), which may be capable of acting on compounds beyond the predicted seed compounds. Thus, we used these annotations to define nutrient sources, rather than using the predicted seed compounds themselves. Among these nutrient sources were di- and polypeptides, predicted to be released from both cytosolic- and membrane-bound aminopeptidases. As discussed below, we identified a number of transport proteins capable of transporting these released residues. In Lake Mendota, these aminopeptidases were expressed in clades acI-A and acI-B at around 70% of the median gene expression levels, while they were expressed at up to twice the median in clade acI-C (Table S8). These findings agrees with MAR-FISH and CARD-FISH studies that confirm the ability of acI bacteria to consume a variety of amino acids [@Salcher2010; @Salcher2013].

All three clades were predicted to encode an alpha-glucosidase, which in Lake Mendota was expressed most strongly in clade acI-C, at approximately 116% of the median (Table S9). Clades acI-A and acI-C also encode a beta-glucosidase, but it was not expressed, at least under prevailing environmental conditions. Both of these enzymes release glucose monomers, which acI is known to consume [@Buck2009; @Salcher2013]. Furthermore, these two clades encode an alpha-galactosidase and multiple maltodextrin glucosidases (which free maltose from maltotriose), but these were only expressed in clade acI-C during our sampling period. The alpha-galactosidase had a log2 average RPKM expression value of 2.5 times the median, while the maltodextrin glucosidases were expressed at approximately 20% of the median (Table S9).

## Compounds Transported by the acI Lineage

Microbes may be capable of transporting compounds that are not strictly required for growth, and comparing such compounds to predicted seed compounds can provide additional information about an organism's ecology. Thus, we used the metabolic network reconstructions for the acI genomes to systematically characterize the transport capabilities of the acI lineage.

All acI clades encode for and expressed a diverse array of transporters (Figure 4, Tables S10 and S11, and the Supplemental Online Material). Consistent with the presence of peptidases, all clades contain numerous genes for the transport of peptides and amino acids, including multiple oligopeptide and branched-chain amino acid transporters, as well as two distinct transporters for the polyamines spermidine and putrescine. All clades also contain a transporter for ammonium. As averaged over the 24-hour sampling period, the ammonium, branched-chain amino acid, and oligopeptide transporters had expression values above the median, with expression values for the substrate-binding protein (of the ATP-binding cassette (ABC) transporters) ranging from 2 to 325 times the median (Table S10). In contrast, while all clades expressed some genes from the polyamine transporters, only clade acI-B expressed the spermidime/putrescine binding protein, at approximately 75 times the median (Table S10). Additionally, clade acI-A contains a third distinct branched-chain amino acid transporter, composed of COGs not found in clades acI-B or acI-C. This transporter was not as highly expressed as the shared transporters, with the substrate-binding protein not expressed at all (Table S10). Finally, clades acI-A and acI-B also contain a transporter for glycine betaine, which was only expressed in clade acI-A, at approximately 35 times the median (Table S10). However, because these observations were made at a single site at a single point in time, we cannot rule out the possibility that the expression of these transporters changes with space and time.

All clades also strongly expressed transporters consistent with the presence of glycoside hydrolases, including transporters for the sugars maltose (a dimer of glucose) and xylose, with expression values for the substrate-binding protein ranging from 3 to 144 times the median (Table S10). Clades acI-A and acI-B also contain four distinct transporters for ribose, although the substrate-binding subunit was not expressed at the time of sampling (Table S10).

Representatives from the acI lineage also encode and expressed a number of transporters that do not have corresponding seed compounds, including a uracil permease, and a xanthine/uracil/thiamine/ascorbate family permease, both of which are expressed at levels ranging from 11 to 127 times the median (Table S10) during the sampling period. Clades acI-A and acI-B also contain a cytosine/purine/uracil/thiamine/allantoin family permease, even if it was only expressed in clade acI-B at the time of sampling (Table S10). Though not strictly annotated as such, all three of these transporters may be responsible for the uptake of the seed compound UMP. In addition, clade acI-A contains but did not express a transporter for cobalamin (Vitamin B12), and both clades acI-A and acI-B contain but did not express transporters for thiamin (Vitamin B1) and biotin (Vitamin B7) (Table S10). Despite predicted auxotrophies for Vitamins B5 and B6, we were unable to find transporters for these two compounds. However, as identification and annotation of transport proteins is an active area of research [@Saier2014], transporters for these vitamins may yet be present in the genomes.

Finally, all three clades expressed actinorhodopsin, a light-sensitive protein that functions as an proton efflux pump [@Sharma2008]. In all clades, actinorhodopsin was among the top seven most highly-expressed genes at the time of sampling (Table S4), with expression values in excess of 300 times the median in all three clades (Table S4). Given that many of the transport proteins are ABC transporters, we speculate that actinorhodopsin may facilitate maintenance of the proton gradient necessary for ATP synthesis. Above-median expression of the ATP synthase genes is consistent with this hypothesis. Coupled with high expression levels of diverse transporters, this result strongly suggests that acI functioned as a photoheterotroph and was actively pumping protons during our sampling period. However, it remains to be seen if this behavior is a general feature of acI ecology or restricted to the specific conditions of the lake and our sampling period.

# Discussion

This study introduces the use of high-throughput metabolic network reconstruction and the seed set framework to predict auxotrophies and nutrient sources of uncultivated microorganisms from incomplete genome sequences. By leveraging multiple genomes from closely related populations, we were able to construct composite genomes for individual acI clades. Obviously this masks differences among tribes as well as smaller populations and individual cells, and may sometimes overestimate the metabolic capabilities of a clade or group. However, it provides a framework that can be used to generate new hypotheses about the substrates used by members of a defined phylogenetic group, provided multiple closely related genomes are available. As metagenomic assembly and binning techniques and single cell sequencing methods improve and complete genomes become available, we anticipate our approach being applied to individual microbial genomes.

Our predictions of substrate use capabilities of the acI lineage are largely congruent with previous genome-based studies based on smaller but manually curated genome collections, indicating that the use of automatic metabolic network reconstructions yields similar predictions to manual metabolic reconstruction efforts, while being both high-throughput and focused on an organism's substrate utilization capabilities. In particular, this study predicts that the consumption of N-rich compounds is a universal feature of the acI lineage, with all three clades predicted to consume ammonium, branched-chain amino acids, and di- and oligopeptides. We provide new evidence for further specialization within each clade, identifying unique substate binding proteins for some of their amino acid and peptide transporters (see Supplemental Online Material), and the expression of a transporter for the polyamines spermidine and putrescine in clade acI-B. Furthermore, we confirm the ability of all three clades to consume xylose and maltose, and of clades acI-A and acI-B to consume ribose. However, despite the presence and expression of alpha-glucosidases in all three clades, and beta-glucosidases in clades acI-A and acI-B, no obvious glucose transport system was found in the genomes. Our analysis also made novel predictions, including the presence of beta-glucosidases, as well as alpha- and beta-galactosidases, in clades acI-A and acI-C.

This study also suggests that auxotrophies for some vitamins may be universal features of the acI lineage, as we predict all clades to be auxotrophic for pantothenic acid and pyridoxal 5'-phosphate (Vitamins B5 and B6). We also predict new auxotrophies within the acI lineage, including THF (clade acI-A), and lysine, homoserine, and UMP (clade acI-C). While our acI-C composite genome remains incomplete, these results nonetheless provide additional support to the hypothesis that distributed metabolic pathways and metabolic complementarity may be common features of freshwater bacterial communities [@Garcia2015; @Garcia2016].

Combined, these results indicate that acI are photoheterotrophs, making a living on a diverse array of N-rich compounds, sugars, oligo- and poly-saccharides, and light. We hypothesize that the acI obtain peptides from the products of cell lysis, and may participate in the turnover of high molecular weight dissolved organic compounds, such as starch, glycogen, and cellulose. The acI lineage does not appear to be metabolically self-sufficient, relying on other organisms for the production of essential nutrients.

This study also presents the first combined genomic and metatranscriptomic analysis of a freshwater microbial lineage. Transport proteins were among the most highly expressed in the acI genomes, and the expression of multiple amino acid transporters may facilitate uptake of these labile compounds. We also observed differences in the relative expression of these transporters, which may point to clade-specific differences in the affinity for these substrates. Finally, the actinorhodopsin protein was highly expressed, and may facilitate synthesis of the ATP needed to drive acI's many ABC-type transporters.

A close comparison of our predictions to previous studies of the acI lineage reveals some important limitations of the seed set framework and automatic metabolic reconstructions. First, the seed set framework only identifies compounds that the metabolic network __must__ obtain from its environment, and will fail to identify compounds that the organism can acquire from its environment but can also synthesize itself. For example, members of clades acI-A and acI-B are capable of consuming branched-chain amino acids [@Ghylin2014; @Garcia2015], but can also synthesize them. Thus, these compounds were not identified as seed compounds. However, transport reactions for branched-chain amino acids were identified in the genomes, and our metatranscriptomic found them to be highly expressed.

Second, automatic metabolic network reconstructions may not fully capture an organism's metabolic network (e.g., due to missing or incorrect genome annotations). For example, previous genome-based studies have suggested that members of the acI lineage harbor cyanophycinase, an enzyme that allows them to hydrolyze the cyanobacterial peptide cyanophycin [@Garcia2013; @Ghylin2014]. Manual inspection revealed that KBase annotated this putative enzyme as a hypothetical protein, and we could not identify transporters for cyanophycin in the metabolic network reconstruction. As biochemical characterization of hypothetical proteins and automatic gene and protein annotation are active areas of research, we anticipate that advances in these fields will continue to improve the accuracy of automatic metabolic network reconstructions.

# Conclusions

In this study, we examined the ecological niche of uncultivated acI bacteria using automatic metabolic network reconstructions and the seed set framework combined with metatranscriptomics. Predicted seed compounds include peptides and saccharides, many of which acI have been observed to consume _in situ_, as well as newly predicted auxotrophies for vitamins and amino acids. Metatranscriptomic analysis in a lake with abundant acI members suggests many of these compounds are consumed by acI bacteria in their natural environment. Our high-throughput approach easily scales to 100s of genomes, and enables a focused metabolic analysis by identifying those compounds through which an organism interacts with its environment. Finally, the seed set framework enables additional reverse ecological analyses, which promise to predict the interactions among microbial species in complex environments [@Levy2012].

# Acknowledgements

We thank past members of the McMahon lab for collecting water samples for single-cell sequencing and metagenomic sequencing. This work was supported through the JGI Community Science Program. The work conducted by the JGI, a DOE Office of Science User Facility, is supported by the Office of Science of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231. This material is based upon work that is supported by the National Institute of Food and Agriculture, U.S. Department of Agriculture, under award number 2016-67012-24709 to JJH and WIS01789 to KDM. KDM also acknowledges funding from the United States National Science Foundation (NSF) Microbial Observatories program (MCB-0702395), the NSF Long Term Ecological Research program (NTL-LTER DEB-1440297), an NSF INSPIRE award (DEB-1344254), and the University of Wisconsin System. KDM and KTF acknowledge National Oceanic and Atmospheric Administration (NOAA) grant #NA10OAR4170070, Wisconsin Sea Grant College Program Project #HCE-25, through NOAA’S National Sea Grant College Program, U.S. Department of Commerce.

# Conflict of Interest

The authors declare no conflict of interest.

# References
