{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reverse Ecology and Metatranscriptomics of Uncultivated Freshwater Actinobacteria"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 1: Phylogenetic Tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Overview\n",
    "Construction of the phylogenetic tree requires the following steps:\n",
    "\n",
    "1. Download reference genomes.\n",
    "2. Extract concatenated marker gene sequences using [Phylosift](https://phylosift.wordpress.com/).\n",
    "3. Construct phylogenetic tree using [RAxML](http://sco.h-its.org/exelixis/web/software/raxml/index.html).\n",
    "4. Visualize phylogenetic tree using the [ETE Toolkit](http://etetoolkit.org/) for Python."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 1: Downloading Reference Genomes\n",
    "\n",
    "The reference genomes used in this study are scattered throughout the web. __Later, I will write code to download the genomes to `data/refGenomes/fna`.__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 2: Extract concatenated marker gene sequences using Phylosift"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%reset\n",
    "# Import packages\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "import subprocess\n",
    "\n",
    "# Define folders\n",
    "genomeDir = '../data/refGenomes/fna'\n",
    "outgroupDir = '../data/refGenomes/outgroups'\n",
    "scratchDir = '../data/scratch'\n",
    "psInputDir = scratchDir+'/psInput'\n",
    "resultsDir = '../figures'\n",
    "treeDir = resultsDir+'/fig1-tree'\n",
    "\n",
    "# Check that psInputDir exists: genomes for use with Phylosift will get\n",
    "# transferred there\n",
    "if not os.path.exists(psInputDir):\n",
    "    os.makedirs(psInputDir)\n",
    "    \n",
    "# Check that resultsDir and treeDir exist: alignment for use with Raxml will get\n",
    "# transferred there\n",
    "if not os.path.exists(resultsDir):\n",
    "    os.makedirs(resultsDir)\n",
    "if not os.path.exists(treeDir):\n",
    "    os.makedirs(treeDir)\n",
    "\n",
    "# Retrieve the list of genomes to process\n",
    "genomeList = []\n",
    "for item in os.listdir(genomeDir):\n",
    "    if item.endswith('.fna'):\n",
    "        genomeList.append(item)\n",
    "        \n",
    "genomeList = [genome.replace('.fna', '') for genome in genomeList]\n",
    "        \n",
    "outgroupList = []\n",
    "for item in os.listdir(outgroupDir):\n",
    "    if item.endswith('.fna'):\n",
    "        outgroupList.append(item)\n",
    "\n",
    "outgroupList = [genome.replace('.fna', '') for genome in outgroupList]\n",
    "\n",
    "# For each genome, concatenate individual contigs. Repeat for refGenomes and outgroups.\n",
    "for genome in genomeList:\n",
    "    inFile = open(genomeDir+'/'+genome+'.fna')\n",
    "    outFile = open(psInputDir+'/'+genome+'.fna', \"w\")\n",
    "    \n",
    "    # Create a header for the fasta file\n",
    "    print >>outFile, '>'+genome\n",
    "    \n",
    "    # Concatenate all the sequences\n",
    "    seqList = []\n",
    "    for seqRecord in SeqIO.parse(inFile, 'fasta'):\n",
    "        seqList.append(seqRecord.seq)\n",
    "        \n",
    "    concatSeq = \"\".join([str(seqRec) for seqRec in seqList])\n",
    "    print >>outFile, concatSeq\n",
    "    \n",
    "    inFile.close()\n",
    "    outFile.close()\n",
    "\n",
    "for genome in outgroupList:\n",
    "    inFile = open(outgroupDir+'/'+genome+'.fna')\n",
    "    outFile = open(psInputDir+'/'+genome+'.fna', \"w\")\n",
    "    \n",
    "    # Create a header for the fasta file\n",
    "    print >>outFile, '>'+genome\n",
    "    \n",
    "    # Concatenate all the sequences\n",
    "    seqList = []\n",
    "    for seqRecord in SeqIO.parse(inFile, 'fasta'):\n",
    "        seqList.append(seqRecord.seq)\n",
    "        \n",
    "    concatSeq = \"\".join([str(seqRec) for seqRec in seqList])\n",
    "    print >>outFile, concatSeq\n",
    "    \n",
    "    inFile.close()\n",
    "    outFile.close()\n",
    "    \n",
    "print \"Processing complete.\"\n",
    "print \"Running phylosift.\"\n",
    "    \n",
    "# Run Phylosift search and alignment steps\n",
    "for genome in genomeList:\n",
    "   print \"Running search step on genome %s (%s of %s)\" % (genome, genomeList.index(genome)+1, len(genomeList))\n",
    "   subprocess.call(['/Applications/phylosift/phylosift', 'search', '--besthit', psInputDir+'/'+genome+'.fna'])\n",
    "   \n",
    "   print \"Running align step on genome %s (%s of %s)\" % (genome, genomeList.index(genome)+1, len(genomeList))\n",
    "   subprocess.call(['/Applications/phylosift/phylosift', 'align', '--besthit', psInputDir+'/'+genome+'.fna'])\n",
    "    \n",
    "for genome in outgroupList:\n",
    "   print \"Running search step on outgroup %s (%s of %s)\" % (genome, outgroupList.index(genome)+1, len(outgroupList))\n",
    "   subprocess.call(['/Applications/phylosift/phylosift', 'search', '--besthit', psInputDir+'/'+genome+'.fna'])\n",
    "   \n",
    "   print \"Running align step on outgroup %s (%s of %s)\" % (genome, outgroupList.index(genome)+1, len(outgroupList))\n",
    "   subprocess.call(['/Applications/phylosift/phylosift', 'align', '--besthit', psInputDir+'/'+genome+'.fna'])\n",
    "   \n",
    "print \"Phylosift complete.\"\n",
    "\n",
    "# Create concatenated alignment file.\n",
    "\n",
    "outFile = open(treeDir+'/'+'tree.aln', \"w\")\n",
    "\n",
    "for genome in genomeList+outgroupList:\n",
    "    inFile = open('PS_temp'+'/'+genome+'.fna'+'/alignDir/concat.updated.1.fasta')\n",
    "    \n",
    "    print >>outFile, '>'+genome\n",
    "    for seqRecord in SeqIO.parse(inFile, 'fasta'):\n",
    "        print >>outFile, seqRecord.seq\n",
    "        \n",
    "    inFile.close()\n",
    "outFile.close()\n",
    "\n",
    "print \"Concatenation complete.\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3: Construct phylogenetic tree using RAxML\n",
    "\n",
    "RAxML will take a long time to run if run on a personal computer. Instead, you can run FastTree to get approximately the same results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%reset\n",
    "# Import packages\n",
    "import os\n",
    "import shutil\n",
    "import subprocess\n",
    "\n",
    "# Define folders\n",
    "resultsDir = '../figures'\n",
    "treeDir = resultsDir+'/fig1-tree'\n",
    "\n",
    "# Run FastTree if you're running this on a personal computer\n",
    "# The command is: /Applications/phylosift/bin/FastTree -out scratch/raxml/tree.nwk scratch/raxml/tree.aln \n",
    "subprocess.call(['/Applications/phylosift/bin/FastTree', '-out', treeDir+'/tree.nwk', \n",
    "                 treeDir+'/tree.aln'])\n",
    "\n",
    "# Actual RAxML command run on a server\n",
    "# The command is: /Applications/RAxML/raxmlHPC-PTHREADS-AVX -f a -m PROTGAMMAAUTO -p 12345 -x 12345 -# 100 -s results/fig1-tree/tree.aln -T 24 -n tree\n",
    "#subprocess.call(['/Applications/RAxML/raxmlHPC-PTHREADS-AVX', '-f', 'a', '-m',  \n",
    "#                 'PROTGAMMAAUTO', '-p', '12345', '-x', '12345', '-#', '100', \n",
    "#                 '-s', treeDir+'/tree.aln', '-T', '24', '-n', 'tree'])\n",
    "\n",
    "# Move the results of the RAxML run to treeDir\n",
    "for myFile in os.listdir('.'):\n",
    "    if '.tree' in myFile:\n",
    "        shutil.move(myFile, treeDir)\n",
    "\n",
    "# Rename RAxML_bipartitions.tree into tree.aln\n",
    "shutil.move(treeDir+'/RAxML_bipartitions.tree', treeDir+'/tree.nwk')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 4: Visualize phylogenetic tree using ETE Toolkit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%reset\n",
    "# Import packages\n",
    "import ete3\n",
    "import re\n",
    "import os\n",
    "\n",
    "# Define folders\n",
    "externalDataDir = '../data/externalData'\n",
    "resultsDir = '../figures'\n",
    "treeDir = resultsDir+'/fig1-tree'\n",
    "\n",
    "# Check that treeDir exists\n",
    "if not os.path.exists(treeDir):\n",
    "    os.makedirs(treeDir)\n",
    "    \n",
    "# Read in the phylogenetic tree\n",
    "actinoTree = ete3.Tree(treeDir+'/tree.nwk')\n",
    "\n",
    "# Set the outgroup\n",
    "actinoTree.set_outgroup('Pnec')\n",
    "\n",
    "# Define the node style\n",
    "def nodeStyleFunct(node):\n",
    "    nodeStyle = ete3.NodeStyle()\n",
    "    nodeStyle['size'] = 5\n",
    "    nodeStyle['shape'] = 'circle'\n",
    "    if node.support >= 90:\n",
    "        nodeStyle['fgcolor'] = 'black'\n",
    "    elif node.support >= 70:\n",
    "        nodeStyle['fgcolor'] = 'grey'\n",
    "    else:\n",
    "        nodeStyle['size'] = 0\n",
    "    # Apply to all nodes\n",
    "    node.set_style(nodeStyle)\n",
    "\n",
    "# Label internal nodes with clade names\n",
    "# Create a dict showing the leaves assoc with each clade\n",
    "cladeNodeDict = {}\n",
    "with open(externalDataDir+'/cladeLeaves.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        line = re.sub('[\"\\n]', '', line)\n",
    "        key = line.split(',')[0]\n",
    "        value = line.split(',')[1:]\n",
    "        cladeNodeDict[key] = value    \n",
    "# Add labels to internal nodes, based on their descendants\n",
    "for key in cladeNodeDict.keys():\n",
    "    for node in actinoTree.traverse():\n",
    "        if set(node.get_leaf_names()) == set(cladeNodeDict[key]):\n",
    "            node.add_face(ete3.TextFace(key, fgcolor='blue'), 0, 'branch-top')\n",
    "\n",
    "# Update internal node names with proper genome/species name\n",
    "# Create a dict mapping node to species names\n",
    "leafNameDict = {}\n",
    "with open(externalDataDir+'/leafNames.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        [key, value] = line.split(',')\n",
    "        leafNameDict[key] = value.strip()\n",
    "# Place genome names at the right edge of the tree and align        \n",
    "for node in actinoTree.iter_leaves():\n",
    "# If node describes a species, make it italic\n",
    "    if len(leafNameDict[node.name].split(' ')) > 1:\n",
    "        node.add_face(ete3.TextFace(leafNameDict[node.name], fstyle='italic'), 0, 'branch-right')\n",
    "    else:\n",
    "        node.add_face(ete3.TextFace(leafNameDict[node.name]), 0, 'branch-right')\n",
    "\n",
    "# Create a column on the right-hand side showing taxonomy of outgroups\n",
    "rhsDict = {}\n",
    "with open(externalDataDir+'/taxonNames.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        [key, value] = line.split(',')\n",
    "        rhsDict[key] = value.strip()\n",
    "# Place genome names at the right edge of the tree and align        \n",
    "for node in actinoTree.iter_leaves():\n",
    "    if node.name in rhsDict.keys():\n",
    "        node.add_face(ete3.TextFace(rhsDict[node.name]), 0, 'aligned')\n",
    "    \n",
    "# Define the tree style\n",
    "treeStyle = ete3.TreeStyle()\n",
    "treeStyle.layout_fn = nodeStyleFunct\n",
    "#treeStyle.show_branch_support = True\n",
    "treeStyle.branch_vertical_margin = 10 # vertical spacing\n",
    "#treeStyle.draw_guiding_lines = True\n",
    "treeStyle.show_leaf_name = False\n",
    "\n",
    "# Render the phylogenetic tree\n",
    "#actinoTree.render(\"%%inline\", tree_style = treeStyle)\n",
    "actinoTree.render(treeDir+'/tree-full.svg', tree_style = treeStyle)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simplify Tree for Manuscript\n",
    "\n",
    "This bit of code simplifies the tree to retain only a single Actinomycetales genome as the outgroup. The figure is then touched up in Adobe Illustrator to give the final figure presented in the mansucript."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%reset\n",
    "# Import packages\n",
    "import ete3\n",
    "import re\n",
    "import os\n",
    "\n",
    "# Define folders\n",
    "externalDataDir = '../data/externalData'\n",
    "resultsDir = '../figures'\n",
    "treeDir = resultsDir+'/fig1-tree'\n",
    "\n",
    "# Check that treeDir exists\n",
    "if not os.path.exists(treeDir):\n",
    "    os.makedirs(treeDir)\n",
    "    \n",
    "# Read in the phylogenetic tree\n",
    "actinoTree = ete3.Tree(treeDir+'/tree.nwk')\n",
    "\n",
    "# Define the node style\n",
    "def nodeStyleFunct(node):\n",
    "    nodeStyle = ete3.NodeStyle()\n",
    "    nodeStyle['size'] = 5\n",
    "    nodeStyle['shape'] = 'circle'\n",
    "    if node.support >= 90:\n",
    "        nodeStyle['fgcolor'] = 'black'\n",
    "    elif node.support >= 70:\n",
    "        nodeStyle['fgcolor'] = 'grey'\n",
    "    else:\n",
    "        nodeStyle['size'] = 0\n",
    "    # Apply to all nodes\n",
    "    node.set_style(nodeStyle)\n",
    "\n",
    "# # Label internal nodes with clade names\n",
    "# # Create a dict showing the leaves assoc with each clade\n",
    "# cladeNodeDict = {}\n",
    "# with open(externalDataDir+'/cladeLeaves.csv', 'rU') as inFile:\n",
    "#     for line in inFile:\n",
    "#         line = re.sub('[\"\\n]', '', line)\n",
    "#         key = line.split(',')[0]\n",
    "#         value = line.split(',')[1:]\n",
    "#         cladeNodeDict[key] = value    \n",
    "# # Add labels to internal nodes, based on their descendants\n",
    "# for key in cladeNodeDict.keys():\n",
    "#     for node in actinoTree.traverse():\n",
    "#         if set(node.get_leaf_names()) == set(cladeNodeDict[key]):\n",
    "#             node.add_face(ete3.TextFace(key, fgcolor='blue'), 0, 'branch-top')\n",
    "\n",
    "# Update internal node names with proper genome/species name\n",
    "# Create a dict mapping node to species names\n",
    "leafNameDict = {}\n",
    "with open(externalDataDir+'/leafNames.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        [key, value] = line.split(',')\n",
    "        leafNameDict[key] = value.strip()\n",
    "# Place genome names at the right edge of the tree and align        \n",
    "for node in actinoTree.iter_leaves():\n",
    "# If node describes a species, make it italic\n",
    "    if len(leafNameDict[node.name].split(' ')) > 1:\n",
    "        node.add_face(ete3.TextFace(leafNameDict[node.name], fstyle='italic'), 0, 'branch-right')\n",
    "    else:\n",
    "        node.add_face(ete3.TextFace(leafNameDict[node.name]), 0, 'branch-right')\n",
    "\n",
    "# Create a column on the right-hand side showing taxonomy of outgroups\n",
    "rhsDict = {}\n",
    "with open(externalDataDir+'/taxonNames.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        [key, value] = line.split(',')\n",
    "        rhsDict[key] = value.strip()\n",
    "# Place genome names at the right edge of the tree and align        \n",
    "for node in actinoTree.iter_leaves():\n",
    "    if node.name in rhsDict.keys():\n",
    "        node.add_face(ete3.TextFace(rhsDict[node.name]), 0, 'aligned')\n",
    "    \n",
    "# Define the tree style\n",
    "treeStyle = ete3.TreeStyle()\n",
    "treeStyle.layout_fn = nodeStyleFunct\n",
    "#treeStyle.show_branch_support = True\n",
    "treeStyle.branch_vertical_margin = 10 # vertical spacing\n",
    "#treeStyle.draw_guiding_lines = True\n",
    "treeStyle.show_leaf_name = False\n",
    "\n",
    "# Drop individual nodes\n",
    "nodeDeleteList = ['Pnec', '637000248', '646311917', '2503538010', '646564571', \n",
    "                  '648028042', '646311931', '2506783060', '639633044', '646564587', \n",
    "                  '646311932', '651053061', '644736393', '2511231200', '644736339', \n",
    "                  '646311938', '646361958', '646311958', \n",
    "                  '648028043', '2585427650', '2639763019', '2576861448', '646564505',\n",
    "                  '2505679089', '646564520', '646311968', '644736376', '646564565',\n",
    "                  '644736331', '640753031', '643348509', '644736322', '639633001', \n",
    "                  '650377905', '643692008', '643348516', '646564566', '2639762896']\n",
    "\n",
    "for node in actinoTree.iter_leaves():\n",
    "    if node.name in nodeDeleteList:\n",
    "        node.delete(preserve_branch_length=True)\n",
    "\n",
    "# Set the outgroup\n",
    "actinoTree.set_outgroup('2684623058')\n",
    "\n",
    "# Render the phylogenetic tree\n",
    "#actinoTree.render(\"%%inline\", tree_style = treeStyle)\n",
    "actinoTree.render(treeDir+'/tree-ms.svg', tree_style = treeStyle, w=9, units='in')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Simplify Tree for Supplement\n",
    "\n",
    "This bit of code simplifies the tree to include only a single genome for each freshwater tribe and for the outgroup classes. The figure is then touched up in Adobe Illustrator to give the final figure presented in the mansucript."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%reset\n",
    "# Import packages\n",
    "import ete3\n",
    "import re\n",
    "import os\n",
    "\n",
    "# Define folders\n",
    "externalDataDir = '../data/externalData'\n",
    "resultsDir = '../figures'\n",
    "treeDir = resultsDir+'/fig1-tree'\n",
    "\n",
    "# Check that treeDir exists\n",
    "if not os.path.exists(treeDir):\n",
    "    os.makedirs(treeDir)\n",
    "    \n",
    "# Read in the phylogenetic tree\n",
    "actinoTree = ete3.Tree(treeDir+'/tree.nwk')\n",
    "\n",
    "# Define the node style\n",
    "def nodeStyleFunct(node):\n",
    "    nodeStyle = ete3.NodeStyle()\n",
    "    nodeStyle['size'] = 5\n",
    "    nodeStyle['shape'] = 'circle'\n",
    "    if node.support >= 90:\n",
    "        nodeStyle['fgcolor'] = 'black'\n",
    "    elif node.support >= 70:\n",
    "        nodeStyle['fgcolor'] = 'grey'\n",
    "    else:\n",
    "        nodeStyle['size'] = 0\n",
    "    # Apply to all nodes\n",
    "    node.set_style(nodeStyle)\n",
    "\n",
    "# # Label internal nodes with clade names\n",
    "# # Create a dict showing the leaves assoc with each clade\n",
    "# cladeNodeDict = {}\n",
    "# with open(externalDataDir+'/cladeLeaves.csv', 'rU') as inFile:\n",
    "#     for line in inFile:\n",
    "#         line = re.sub('[\"\\n]', '', line)\n",
    "#         key = line.split(',')[0]\n",
    "#         value = line.split(',')[1:]\n",
    "#         cladeNodeDict[key] = value    \n",
    "# # Add labels to internal nodes, based on their descendants\n",
    "# for key in cladeNodeDict.keys():\n",
    "#     for node in actinoTree.traverse():\n",
    "#         if set(node.get_leaf_names()) == set(cladeNodeDict[key]):\n",
    "#             node.add_face(ete3.TextFace(key, fgcolor='blue'), 0, 'branch-top')\n",
    "\n",
    "# Update internal node names with proper genome/species name\n",
    "# Create a dict mapping node to species names\n",
    "leafNameDict = {}\n",
    "with open(externalDataDir+'/leafNames.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        [key, value] = line.split(',')\n",
    "        leafNameDict[key] = value.strip()\n",
    "# Place genome names at the right edge of the tree and align        \n",
    "for node in actinoTree.iter_leaves():\n",
    "# If node describes a species, make it italic\n",
    "    if len(leafNameDict[node.name].split(' ')) > 1:\n",
    "        node.add_face(ete3.TextFace(leafNameDict[node.name], fstyle='italic'), 0, 'branch-right')\n",
    "    else:\n",
    "        node.add_face(ete3.TextFace(leafNameDict[node.name]), 0, 'branch-right')\n",
    "\n",
    "# Create a column on the right-hand side showing taxonomy of outgroups\n",
    "rhsDict = {}\n",
    "with open(externalDataDir+'/taxonNames.csv', 'rU') as inFile:\n",
    "    for line in inFile:\n",
    "        [key, value] = line.split(',')\n",
    "        rhsDict[key] = value.strip()\n",
    "# Place genome names at the right edge of the tree and align        \n",
    "for node in actinoTree.iter_leaves():\n",
    "    if node.name in rhsDict.keys():\n",
    "        node.add_face(ete3.TextFace(rhsDict[node.name]), 0, 'aligned')\n",
    "    \n",
    "# Define the tree style\n",
    "treeStyle = ete3.TreeStyle()\n",
    "treeStyle.layout_fn = nodeStyleFunct\n",
    "#treeStyle.show_branch_support = True\n",
    "treeStyle.branch_vertical_margin = 10 # vertical spacing\n",
    "#treeStyle.draw_guiding_lines = True\n",
    "treeStyle.show_leaf_name = False\n",
    "\n",
    "# Drop individual nodes\n",
    "nodeDeleteList = ['Pnec', '637000248', '646311917', '2503538010', '646564571', \n",
    "                  '648028042', '646311931', '2506783060', '639633044', '646564587', \n",
    "                  '646311932', '651053061', '644736393', '2511231200', '644736339', \n",
    "                  '646311938', '646361958', '646311958', \n",
    "                  '648028043', '2585427650', '2639763019', '2576861448', '646564505',\n",
    "                  '2505679089', '646564520', '646311968', '644736376', '646564565',\n",
    "                  '644736331', '640753031', '643348509',\n",
    "                 'MEint885', 'BIN_10',\n",
    "                 'AAA278O22', 'AAA023J06', 'AAA024D14', 'AAA041L13', 'AAA028E20', 'AAA044O16',\n",
    "                 'AAA044D11', 'acIB-AMD-6', 'AAA028A23', 'AAA027L06', 'AB141P03', 'AAA027J17',\n",
    "                  'AAA023D18', 'AAA278I18', 'MEint283']\n",
    "\n",
    "for node in actinoTree.iter_leaves():\n",
    "    if node.name in nodeDeleteList:\n",
    "        node.delete(preserve_branch_length=True)\n",
    "\n",
    "# Set the outgroup\n",
    "actinoTree.set_outgroup('644736322')\n",
    "\n",
    "# Render the phylogenetic tree\n",
    "#actinoTree.render(\"%%inline\", tree_style = treeStyle)\n",
    "actinoTree.render(treeDir+'/tree-suppl.svg', tree_style = treeStyle, w=9, units='in')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
